File: common/src/main/java/gollorum/signpost/utils/CollectionUtils.java
package gollorum.signpost.utils;

import it.unimi.dsi.fastutil.objects.Object2IntLinkedOpenHashMap;
import it.unimi.dsi.fastutil.objects.Object2IntMap;

import java.util.Collection;
import java.util.Map;

public class CollectionUtils {

	public static <T> Map<T, Integer> group(Collection<T> in) {
		Object2IntMap<T> map = new Object2IntLinkedOpenHashMap<>();
		for(T t : in) map.computeInt(t, (x, i) -> i == null ? 1 : i + 1);
		return map;
	}

}

File: src/main/java/gollorum/signpost/util/collections/CollectionUtils.kt
package gollorum.signpost.util.collections

import java.util.function.Predicate

class CollectionUtils {

    companion object {
        fun <T>find(list : List<T>, where : Predicate<T>) : T? {
            for(t in list){
                if(where.test(t)) return t;
            }
            return null;
        }
    }

}